src/android/java/com/pspdfkit/cordova/action/annotation/AddAnnotationAction.java

```diff
@@ -27,13 +29,21 @@ public AddAnnotationAction(@NonNull String name, @NonNull PSPDFKitCordovaPlugin
   @Override
   protected void execAction(JSONArray args, CallbackContext callbackContext) throws JSONException {
     String annotationJson = args.getJSONObject(ARG_ANNOTATION_JSON).toString();
-    final PdfDocument document = CordovaPdfActivity.getCurrentActivity().getDocument();
+    CordovaPdfActivity pdfActivity = CordovaPdfActivity.getCurrentActivity();
+
+    final PdfDocument document = pdfActivity.getDocument();
     if (document != null) {
-      AnnotationProvider annotationProvider = document.getAnnotationProvider();
-      annotationProvider.addAnnotationToPage(
-          annotationProvider.createAnnotationFromInstantJson(annotationJson)
-      );
+      Annotation annotationFromInstantJson = document.getAnnotationProvider()
+          .createAnnotationFromInstantJson(annotationJson);
```

Comment:
Create annotation also performs annotation adding resulting in `annotationProvider.addAnnotationToPage` being redundant.